File: sdk/AMQP/PHP/request-reply/consumer.php
<?php
/**
* File: consumer.php
* Description: This is the AMQP consumer handles incoming
*     communication from clients publishing messages to a broker server.
*     Messages can be received over AMQP exchange types including one-to-one,
*     from broadcast pattern, or selectively using specified binding key.
*
* Author: Eamin Zhang
* robomq.io (http://www.robomq.io)
*/

require_once __DIR__."/../vendor/autoload.php";
use PhpAmqpLib\Connection\AMQPConnection;
use PhpAmqpLib\Message\AMQPMessage;

$GLOBALS["channel"] = $channel;
$GLOBALS["exchangeName"] = $exchangeName;

$server = "hostname";
$port = 5672;
$vhost = "yourvhost";
$username = "username";
$password = "password";
$exchangeName = "testEx";
$requestQueue = "requestQ";
$requestKey = "request";

//callback funtion on receiving request messages, reply to the reply_to header
$onMessage = function ($message) {
	echo $message->body.PHP_EOL;
	try {
		$replyMessage = new AMQPMessage("Reply to ".$message->body, array("content_type" => "text/plain", "delivery_mode" => 1));
		$GLOBALS["channel"]->basic_publish($replyMessage, $GLOBALS["exchangeName"], $message->get("reply_to"));
		$GLOBALS["channel"]->basic_ack($message->delivery_info["delivery_tag"]);
	} catch (Exception $e) {
		$GLOBALS["channel"]->basic_nack($message->delivery_info["delivery_tag"]);
	}
};

while (true) {
	try {
		//connect
		$connection = new AMQPConnection($server, $port, $username, $password, $vhost);
		$channel = $connection->channel();

		//declare exchange and queue, bind them and consume messages
		$channel->exchange_declare($exchangeName, $type = "direct", false, false, $auto_delete = true);
		$channel->queue_declare($requestQueue, false, false, $exclusive = true, $auto_delete = true);
		$channel->queue_bind($requestQueue, $exchangeName, $requestKey);
		$channel->basic_consume($requestQueue, "", false, $no_ack = false, false, false, $callback = $onMessage);

		//start consuming
		while(count($channel->callbacks)) {
			$channel->wait();
		}
	} catch(Exception $e) {
		//reconnect on exception
		echo "Exception handled, reconnecting...\nDetail:\n".$e.PHP_EOL;
		if ($connection != null) {
			try {
				$connection->close();
			} catch (Exception $e1) {}
		}
		sleep(5);
	}
}
?>

File: sdk/AMQP/Python/request-reply/consumer.py
# File: consumer.py
# Description: This is the AMQP consumer handles incoming
#     communication from clients publishing messages to a broker server.
#     Messages can be received over AMQP exchange types including one-to-one,
#     from broadcast pattern, or selectively using specified binding key.
#
# Author: Stanley
# robomq.io (http://www.robomq.io)

import pika
import time

server = "hostname"
port = 5672
vhost = "yourvhost"
username = "username"
password = "password"
exchangeName = "testEx"
requestQueue = "requestQ"
requestKey = "request"

#callback funtion on receiving request messages, reply to the reply_to header
def onMessage(channel, method, properties, body):
	print body
	try:
		replyProp = pika.BasicProperties(content_type = "text/plain", delivery_mode = 1)
		channel.basic_publish(exchange = exchangeName, routing_key = properties.reply_to, properties = replyProp, body = "Reply to %s" % (body))
		channel.basic_ack(delivery_tag = method.delivery_tag)
	except:
		channel.basic_nack(delivery_tag = method.delivery_tag)
		
while True:
	try:
		#connect
		credentials = pika.PlainCredentials(username, password)
		connection = pika.BlockingConnection(pika.ConnectionParameters(host = server, port = port, virtual_host = vhost, credentials = credentials))
		channel = connection.channel()

		#declare exchange and queue, bind them and consume messages
		channel.exchange_declare(exchange = exchangeName, exchange_type = "direct", auto_delete = True)
		channel.queue_declare(queue = requestQueue, exclusive = True, auto_delete = True)
		channel.queue_bind(exchange = exchangeName, queue = requestQueue, routing_key = requestKey)
		channel.basic_consume(consumer_callback = onMessage, queue = requestQueue, no_ack = False)
		channel.start_consuming()
	except Exception, e:
		#reconnect on exception
		print "Exception handled, reconnecting...\nDetail:\n%s" % e
		try:
			connection.close()
		except:
			pass
		time.sleep(5)